datasets/sentinel-5p/sentinel_5p.py

```diff
@@ -64,135 +18,43 @@
 logger.setLevel(logging.INFO)
 
 
-def recursive_round(coordinates: List[Any], precision: int) -> List[Any]:
-    """Rounds a list of numbers. The list can contain additional nested lists
-    or tuples of numbers.
-
-    Any tuples encountered will be converted to lists.
-
-    Args:
-        coordinates (List[Any]): A list of numbers, possibly containing nested
-            lists or tuples of numbers.
-        precision (int): Number of decimal places to use for rounding.
-
-    Returns:
-        List[Any]: The list of numbers rounded to the given precision.
-    """
-    rounded: List[Any] = []
-    for value in coordinates:
-        if isinstance(value, (int, float)):
-            rounded.append(round(value, precision))
-        else:
-            rounded.append(recursive_round(list(value), precision))
-    return rounded
-
-
-class Sentinel5pNetCDFCollection(Collection):  # type: ignore
+class Sentinel5pNetCDFCollection(Collection):
     @classmethod
     def create_item(
         cls, asset_uri: str, storage_factory: StorageFactory
     ) -> Union[List[pystac.Item], WaitTaskResult]:
 
-        storage, json_path = storage_factory.get_storage_for_file(asset_uri)
-        item_dict = storage.read_json(json_path)
-
-        netcdf_filename = item_dict["assets"]["data"]["href"]
-        match = FILENAME_EXPR.match(Path(netcdf_filename).stem)
-        if not match:
-            raise ValueError(f"Could not parse filename {Path(json_path).stem}")
-
-        prefix = match.groupdict()["product"].strip("_").lower()
-        if prefix.startswith("np"):
-            prefix = prefix.replace("_", "")
-
-        collection_identifier = match.groupdict()["collection"]
-        product_type = item_dict["properties"]["s5p:product_type"]
-        product_name = product_type.lstrip("L2_").rstrip("_").lower().replace("_", "-")
+        storage, nc_path = storage_factory.get_storage_for_file(asset_uri)
 
-        # ---- PROPERTIES ----
-        properties = item_dict.pop("properties")
+        with TemporaryDirectory() as tmp_dir:
+            nc_name = os.path.basename(nc_path)
+            tmp_nc_path = os.path.join(tmp_dir, nc_name)
+            storage.download_file(nc_path, tmp_nc_path)
+            item = create_item(tmp_nc_path)
```

Comment:
It looks like the `href` of (at least the item at https://pct-apis-staging.westeurope.cloudapp.azure.com/stac/collections/sentinel-5p-l2-netcdf/items) is being set to this temporary file.

```python
            "assets": {
                "so2": {
                    "href": "/tmp/tmpir4fdjy7/S5P_NRTI_L2__SO2____20230630T182849_20230630T183349_29600_03_020401_20230630T190605.nc",
                    "type": "application/x-netcdf",
                    "roles": [
                        "data"
                    ],
                    "title": "Sulphur Dioxide Total Column"
                }
            },
```

We'll want to set it to the href. I can take care of that quick.

Reply:
eb5324fd76a3033e5ed60b764309a00800a63934